resources/qml/PrinterSelector/MachineSelector.qml

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2018 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.7
+import QtQuick.Controls 2.3
+
+import UM 1.2 as UM
+import Cura 1.0 as Cura
+
+Cura.ExpandableComponent
+{
+    id: machineSelector
+
+    property bool isNetworkPrinter: Cura.MachineManager.activeMachineNetworkKey != ""
+    property bool isPrinterConnected: Cura.MachineManager.printerConnected
+    property var outputDevice: Cura.MachineManager.printerOutputDevices.length >= 1 ? Cura.MachineManager.printerOutputDevices[0] : null
+
+    popupPadding: UM.Theme.getSize("default_lining").width
+    popupAlignment: Cura.ExpandableComponent.PopupAlignment.AlignLeft
+    iconSource: expanded ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_left")
+
+    UM.I18nCatalog
+    {
+        id: catalog
+        name: "cura"
+    }
+
+    headerItem: Cura.IconLabel
+    {
+        text: isNetworkPrinter ? Cura.MachineManager.activeMachineNetworkGroupName : Cura.MachineManager.activeMachineName
+        source:
+        {
+            if (isNetworkPrinter)
+            {
+                if (machineSelector.outputDevice != null && machineSelector.outputDevice.clusterSize > 1)
+                {
+                    return UM.Theme.getIcon("printer_group")
+                }
+                return UM.Theme.getIcon("printer_single")
+            }
+            return ""
+        }
+        font: UM.Theme.getFont("medium")
+        color: UM.Theme.getColor("text")
+        iconSize: UM.Theme.getSize("machine_selector_icon").width
+
+        UM.RecolorImage
+        {
+            id: icon
+
+            anchors.bottom: parent.bottom
+            x: UM.Theme.getSize("thick_margin").width
+
+            source: UM.Theme.getIcon("printer_connected")
+            width: UM.Theme.getSize("printer_status_icon").width
+            height: UM.Theme.getSize("printer_status_icon").height
+
+            sourceSize.width: width
+            sourceSize.height: height
+
+            color: UM.Theme.getColor("primary")
+            visible: isNetworkPrinter && isPrinterConnected
+
+            // Make a themable circle in the background so we can change it in other themes
+            Rectangle
+            {
+                id: iconBackground
+                anchors.centerIn: parent
+                // Make it a bit bigger so there is an outline
+                width: parent.width + 2
```

Comment:
Shouldn't the +2 be a certain margin size? Like the lining?

Reply:
Done in 63fab9f038e8615432a8c592138e915351131386